addons/hr_timesheet/static/src/js/task_with_hours.js

```diff
@@ -19,6 +19,14 @@ const TaskWithHours = FieldMany2One.extend({
     _getDisplayNameWithoutHours: function (value) {
         return value && value.split('\u00A0')[0];
     },
+    /**
+    * @private
+    * @param {string} value
+    * @returns {string} display_name
+    */
+    _getDisplayName: function (value) {
+        return value && value.split('\n')[0];
+    },
```

Comment:
Why did you need to add this method?

Reply:
I add this method because _onInputKeyup method of web calls _getDisplayName [line](https://github.com/odoo/odoo/blob/master/addons/web/static/src/legacy/js/fields/relational_fields.js#L870) After enabling task creation from timer sometimes it gets undefined m2o_value and [_getDisplayName](https://github.com/odoo/odoo/blob/master/addons/web/static/src/legacy/js/fields/relational_fields.js#L410) of web don't check the value before calling split function and throws an [error](https://www.awesomescreenshot.com/image/21865994?key=26cf1d6db35d450d873b5c7da10bf776) that's why I add this method and check value before calling split function. Should I check the value in the web _getDisplayName method instead of adding this method?

Reply:
Sorry, but I don't understand why `this.m2o_value` is something equal to undefined. 🤔

Reply:
Timer header task field create depending arguments of Timer when we start the timer we get undefined this.m2o_value and whenever the second argument of the timer is undefined to fix that I add an empty string for taskID field when there is no second argument in timer [commit](https://github.com/odoo/enterprise/pull/23284/commits/5cf0d47342e22304905c69c30064075c427381e1) and remove this function.

Reply:
I cannot reproduce the bug. Can you share me a video?

Reply:
I don't know the functional cause of this bug but sometimes it's reproducible [Video](https://rb.gy/bm6xfa) and most time it's not. :confused:

Reply:
Thanks 🙂